src/dmd/cppmanglewin.d

```diff
@@ -684,22 +684,148 @@ private:
             buf.writestring("?1");
             return;
         }
+        else if (sym.ident)
+        {
+            if (sym.ident.equals(Id.assign))
+            {
+                buf.writestring("?4");
+                return;
+            }
+            else if (sym.ident.equals(Id.eq))
+            {
+                buf.writestring("?8");
+                return;
+            }
+            else if (sym.ident.equals(Id.index))
+            {
+                buf.writestring("?A");
+                return;
+            }
+            else if (sym.ident.equals(Id.call))
+            {
+                buf.writestring("?R");
+                return;
+            }
+        }
         if (TemplateInstance ti = sym.isTemplateInstance())
         {
+            auto id = ti.tempdecl.ident;
+            const(char)* symName = id.toChars();
+
+            int firstTemplateArg = 0;
+
+            // test for special symbols
+            import std.algorithm : countUntil;
+            auto operators = [Id._cast, Id.assign, Id.eq, Id.index, Id.call, Id.opUnary, Id.opBinary, Id.opOpAssign];
+            sizediff_t whichOp = operators.countUntil!((o, n) => n.equals(o))(ti.name);
+            switch (whichOp)
+            {
+                case -1:
+                    break;
+                case 0: // opCast
+                    buf.writestring("?B");
+                    return;
+                case 1: // opAssign
+                    symName = "?4";
+                    break;
+                case 2: // opEquals
+                    symName = "?8";
+                    break;
+                case 3: // opIndex
+                    symName = "?A";
+                    break;
+                case 4: // opCall
+                    symName = "?R";
+                    break;
+                case 5:
+                ..
+                case 7: // opUnary, opBinary, opOpAssign
+                    if (ti.tiargs.dim < 1)
+                        break; // perhaps this is assert? those operators shouldn't have 0 args!
+                    TemplateDeclaration td = ti.tempdecl.isTemplateDeclaration();
+                    assert(td);
+                    TemplateParameter tp = (*td.parameters)[0];
+                    TemplateValueParameter tv = tp.isTemplateValueParameter();
+                    if (!tv || !tv.valType.isString())
+                        break; // expecting a string argument to operators!
+                    Expression exp = (*ti.tiargs)[0].isExpression();
+                    StringExp str = exp.toStringExp();
+                    switch (whichOp)
+                    {
+                        case 5: // opUnary
+                            switch (str.peekSlice)
+                            {
+                                case "*":   symName = "?D";     goto continue_template;
```

Comment:
`case TOK.mul:`

Reply:
`exp.op == TOK.string_`, it doesn't distinguish operators. The value of the string is the operator on question.